src/java.base/linux/classes/sun/nio/ch/EventFD.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (c) 2021, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package sun.nio.ch;
+
+import java.io.IOException;
+
+/*
+ * Provides access to the Linux eventfd object.
+ */
+final class EventFD {
+    private final int efd;
+
+    /**
+     * Creates a blocking eventfd object with initial value zero.
+     */
+    EventFD() throws IOException {
+        efd = eventfd0();
+    }
+
+    /**
+     * Creates an eventfd object with initial value zero and optional
+     * blocking.
+     *
+     * @param blocking Whether the eventfd object is blocking
+     */
+    EventFD(boolean blocking) throws IOException {
+        efd = eventfd0();
+        IOUtil.configureBlocking(IOUtil.newFD(efd), blocking);
```

Comment:
This looks okay but the addition of the blocking parameter means you can go back to one of the early iterations and just calling eventfd with the EFD_NONBLOCK. Up to you.

Reply:
I am not sure I understand this comment as a previous comment preferred `IOUtil.configureBlocking()` to set the blocking state.

Reply:
The previous comments were in the context of the no-arg EventFD constructor where we expected it would be created in blocking mode. In that context EPollSelectorImpl would use IOUtil.configureBlocking to configure it to non-blocking. In the new version, there is a blocking parameter so EventFD can use IOUtil.configureBlocking when blocking is "false", or specify EFD_NONBLOCK to eventfd, either is okay with me.